mkosi/manifest.py

```diff
@@ -231,11 +244,13 @@ def as_dict(self) -> dict[str, Any]:
 
         return {
             # Bump this when incompatible changes are made to the manifest format.
-            "manifest_version": 1,
+            "manifest_version": 2,
```

Comment:
This is not an incompatible change to the manifest format so no need to bump the manifest version. Users simply need to check if the extensions field is in the manifest or not.

Reply:
Makes sense. Done.

Reply:
I think we should bump it nevertheless, because for prospective users of this, it us much easier to reason about "version 1 has these keys", "version 2 has the keys from v1 and this new key" compared to "version 1 has these keys and possibly any other from this set". We should make it easy to consume this. 

Assuming we wouldn't bump it on adding a new key, we would need to bump the version number anyway if we change any of the optional keys. I think it's therefore easier to also bump it on adding one and making a better guarantee about which keys exist.

Reply:
What I meant by optional was in the context of the versioning of keys, i.e. the key can be part of a v1 manifest or not, not the Python type.

Reply:
If you bump the manifest version because you add new, optional fields, every tool parsing the manifest and checking the manifest number needs to get adjusted only for the version check,  not for anything else. I don't think that's a good idea.
I think the manifest version should be handled like sonames+symbol versions of shared libraries: don't change the number if only new keys got added but nothing changed incompatible (so no keys removed, no meaning of a key changed). Tools should be able to scope and ignore new json entries, at least I hope.
If json parsers really suddenly fail if there is a new key, then we need to bump the manifest version.